### PR TITLE
Adding integration test for null handling from realtime source

### PR DESCRIPTION
**Pull request by icefury71**
## Description
Adding an end-end integration test for creating a Pinot table with Kafka records including null values and running queries against this table (with the IS NOT NULL predicate). Related to #4230

## Upgrade Notes
Does this PR prevent a zero down-time upgrade? (Assume upgrade order: Controller, Broker, Server, Minion)
No

Does this PR fix a zero-downtime upgrade introduced earlier?
No

Does this PR otherwise need attention when creating release notes?
No

### CODE REVIEW

File: pinot-spi/src/main/java/org/apache/pinot/spi/utils/builder/TableConfigBuilder.java [hunk 1]
```diff
@@ -255,6 +256,11 @@ public TableConfigBuilder setSegmentPartitionConfig(SegmentPartitionConfig segme
     return this;
   }
 
+  public TableConfigBuilder setNullHandlingEnabledConfig(boolean nullHandlingEnabled) {
```
Comment (Jackie-Jiang) [inline pinot-spi/src/main/java/org/apache/pinot/spi/utils/builder/TableConfigBuilder.java:259]:
```suggestion
  public TableConfigBuilder setNullHandlingEnabled(boolean nullHandlingEnabled) {
```

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/NullHandlingIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+  @Override
+  protected long getCountStarResult() {
+    return 100;
+  }
+
+  @Test
+  public void testTotalCount()
+      throws Exception {
+    String pqlQuery = "SELECT count(*) FROM " + getTableName();
+    String sqlQuery = "SELECT count(*) FROM " + getTableName();
```
Comment (Jackie-Jiang) [inline pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/NullHandlingIntegrationTest.java:130]: (nit) Simplify into the same string?
```suggestion
    String query = "SELECT count(*) FROM " + getTableName();
```

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/BaseClusterIntegrationTest.java [hunk 3]
```diff
@@ -241,6 +242,11 @@ protected IngestionConfig getIngestionConfig() {
     return null;
   }
 
+  @Nullable
```
Comment (Jackie-Jiang) [inline pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/BaseClusterIntegrationTest.java:245]: Remove this annotation

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/NullHandlingIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+  }
+
+  @Override
+  @Nullable
+  protected List<String> getBloomFilterColumns() {
+    return null;
+  }
+
+  @Override
+  @Nullable
```
Comment (Jackie-Jiang) [inline pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/NullHandlingIntegrationTest.java:116]: Remove this annotation

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/BaseClusterIntegrationTest.java [hunk 5]
```diff
@@ -241,6 +242,10 @@ protected IngestionConfig getIngestionConfig() {
     return null;
   }
 
+  protected boolean getNullHandlingEnabledConfig() {
```
Comment (Jackie-Jiang) [inline pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/BaseClusterIntegrationTest.java:245]: (nit) 
```suggestion
  protected boolean getNullHandlingEnabled() {
```
